tests/Language/Java/InlineSpec.hs

```diff
@@ -52,6 +56,11 @@ spec = do
           _ :: JJObject <- [java| $obj1 |]
           return ()
 
+        it "Supports parameterized type synonyms" $ do
+          obj :: List ObjectClass <- [java| new Object() {} |]
```

Comment:
Neither inline-java nor jvm do a dynamic check: it's entirely up to the programmer to be honest about his/her type ascriptions. In this test, we don't care about what the object actually is: the object is only a vehicle for giving it a fancy type including a parameterized type synonym.